src/soc/ibm/power9/istep_14_2.c

```diff
@@ -0,0 +1,51 @@
+#include <cpu/power/istep_14.h>
+#include <cpu/power/istep_13.h>
+
+static void thermalInit(void)
+{
+	for (size_t MCSIndex = 0; MCSIndex < MCS_PER_PROC; ++MCSIndex) {
+		for (size_t MCAIndex = 0; MCAIndex < MCA_PER_MCS; ++MCAIndex) {
+				mca_and_or(
+					mcs_ids[MCSIndex], MCAIndex, MCA_MBA_FARB3Q,
+					~PPC_BITMASK(0, 45),
+					PPC_BIT(10) | PPC_BIT(25) | PPC_BIT(37));
+		}
+		scom_and_for_chiplet(mcs_ids[MCSIndex], MCS_MCMODE0, PPC_BIT(21));
+	}
+}
+
+static void progMCMode0(chiplet_id_t MCSTTarget)
+{
+	uint64_t l_scomMask =
+		PPC_BIT(MCS_MCMODE0_DISABLE_MC_SYNC)
+		| PPC_BIT(MCS_MCMODE0_DISABLE_MC_PAIR_SYNC);
+	uint64_t l_scomData =
+		PPC_BIT(MCS_MCMODE0_DISABLE_MC_SYNC)
+		| PPC_BIT(MCS_MCMODE0_DISABLE_MC_PAIR_SYNC);
+	scom_and_or_for_chiplet(
+		MCSTTarget, MCS_MCMODE0,
+		~l_scomMask, l_scomData & l_scomMask);
+}
+
+static void progMaster(chiplet_id_t MCSTarget)
+{
+	scom_and_for_chiplet(MCSTarget, MCS_MCSYNC, ~PPC_BIT(MCS_MCSYNC_SYNC_GO_CH0));
+	scom_and_or_for_chiplet(MCSTarget, MCS_MCSYNC, ~PPC_BIT(SUPER_SYNC_BIT), PPC_BITMASK(0, 16));
+	scom_and_for_chiplet(MCSTarget, MCS_MCSYNC, ~PPC_BIT(MBA_REFRESH_SYNC_BIT));
+}
+
+static void throttleSync(void)
+{
+	for(size_t MCSIndex = 0; MCSIndex < MCS_PER_PROC; ++MCSIndex)
+	{
+		progMCMode0(mcs_ids[MCSIndex]);
+	}
+	progMaster(mcs_ids[0]);
```

Comment:
Are you sure this is always for MCS0 only?

Reply:
It depends on `ATTR_CHIP_UNIT_POS` and it is always 0 if my reasoning is correct. This attribute is only set in [hb_customized_attrs.xml:75](https://github.com/open-power/hostboot/blob/2c5e34fb129ea4adc85ba707d7fdea7ec694bafb/src/usr/targeting/common/xmltohb/hb_customized_attrs.xml#L75).
In any case, only one `MCS` will be configured by `progMaster()`. The code is a bit strange, but in a nutshell it does just that.
After some simplification it looks like this. You can compare it to [p9_throttle_sync.C#L415](https://github.com/open-power/hostboot/blob/2c5e34fb129ea4adc85ba707d7fdea7ec694bafb/src/import/chips/p9/procedures/hwp/nest/p9_throttle_sync.C#L415)
```
uint8_t l_numMasterProgrammed = 0;
for (auto l_mc : i_mcTargets)
{
    FAPI_ATTR_GET(fapi2::ATTR_CHIP_UNIT_POS, l_mc, l_pos);
    l_sideNum = l_pos / MAX_MC_SIDES_PER_PROC;
    l_mcSide[l_sideNum].masterMc = l_mc;
}
for (l_sideNum = 0; l_sideNum < MAX_MC_SIDES_PER_PROC; l_sideNum++)
{
    if (l_numMasterProgrammed == 0)
    {
        progMaster(l_mcSide[l_sideNum].masterMc);
        l_numMasterProgrammed++;
    }
}
```